lib/config.py

```diff
@@ -98,56 +99,132 @@ def _setup_config_parser_options(self):
         self.parser.add_argument('--verbose', '-v',
                                  help='Set the scripts to be verbose',
                                  action='store_true')
-        self.parser.add_argument('--result-dir', '-r',
-                                 help='Directory to save the RPMs.',
-                                 default='./result')
-        self.parser.add_argument('--repositories-path', '-R',
-                                 help='Directory where to clone code repositories',
-                                 default='/var/lib/host-os/repositories')
         self.parser.add_argument('--keep-builddir',
                                  help='Keep build directory and its logs and '
                                  'artifacts.', action='store_true')
-        self.parser.add_argument('--build-versions-repository-url',
-                                 help='Build versions repository URL')
-        self.parser.add_argument('--build-version',
-                                 help='Select build version from versions '
-                                 'repository')
-        self.parser.add_argument('--build-versions-repo-dir',
-                                 help='Directory to clone the build versions '
-                                 'repository',
-                                 default='./components')
         self.parser.add_argument('--log-size',
                                  help='Size in bytes above which the log file '
                                  'should rotate', type=int)
-        self.parser.add_argument('--http-proxy',
-                                 help='HTTP proxy URL')
-        self.parser.add_argument('--release-notes-repo-url',
-                                 help='Release notes repository URL')
-        self.parser.add_argument('--release-notes-repo-branch',
-                                 help='Branch of the release notes repository '
-                                 'to checkout')
-        self.parser.add_argument('--push-repo-url',
-                                 help='URL of the repository used for pushing')
-        self.parser.add_argument('--push-repo-branch',
-                                 help='Branch of the repository used for '
-                                 'pushing',
-                                 default='master')
-        self.parser.add_argument('--committer-name',
-                                 help='Name used when creating a commit and '
-                                 'bumping spec files')
-        self.parser.add_argument('--committer-email',
-                                 help='Email used when creating a commit and '
-                                 'bumping spec files')
 
+        subparser = self.parser.add_subparsers(
+            dest="subcommand",
+            help="Available subcommands")
+
+        # create the parser for the "build-package" subcommand
+        parser_package = subparser.add_parser('build-package',
```

Comment:
What if we had a list of argument groups like so:
```
build_repo = [
    dict(option='--build-versions-repository-url', help='Build versions repository URL')
    dict(option='--build-versions-repo-dir', help='Directory to clone the build versions repository', default='./components')
]
push_repo = [...]
```

And then made each command 'subscribe' to them:
```
SUBCOMMANDS = [
    ('build-package', 'Build packages', [build_repo] ),
    ('release-notes', 'Create release notes', [push_repo, buld_repo]),
    ...
]
```

```
def add_args():
    for command, help, arg_groups in SUBCOMMANDS:
        subparser = subparser.add_parser(command, help)
        for arg_group in arg_groups:    
            for arg in arg_group:
                subparser.add_argument(**arg)
```